classification/classification_network.py

```diff
@@ -246,109 +220,368 @@ def createClassifierRegion(network, classifierType, classifierParams):
+
+  # keep track of the regions that have been trained
+  trainedRegionNames = []
+  numCorrect = 0
+  numTestRecords = 0
+  for recordNumber in xrange(numRecords):
+    # Run the network for a single iteration
+    network.run(1)
+
+    actualValue = sensorRegion.getOutputData("categoryOut")[0]
+
+    partitionName = partitions[0][0]
+    partitionIndex = partitions[0][1]
+
+    # train all of the regions
+    if partitionIndex == recordNumber:
+
+      if partitionName == TEST_PARTITION_NAME:
+        _stopLearning(network, trainedRegionNames, recordNumber)
+
+      else:
+        partitions.pop(0)
+        trainedRegionNames.append(partitionName)
+        _enableRegionLearning(network,
+                              trainedRegionNames,
+                              partitionName,
+                              recordNumber)
+
+    # Evaluate the predictions on the test set.
+    if recordNumber >= testIndex:
+      if classifierRegion.type == KNN_CLASSIFIER_TYPE:
+        inferredValue = classifierRegion.getOutputData("categoriesOut").argmax()
+      elif classifierRegion.type == CLA_CLASSIFIER_TYPE:
+        inferredValue = classifierRegion.getOutputData("categoriesOut")[0]
+      if actualValue == inferredValue:
+        numCorrect += 1
+      elif DEBUG_VERBOSITY > 0:
+        print ("[DEBUG] recordNum=%s, actualValue=%s, inferredValue=%s"
+               % (recordNumber, actualValue, inferredValue))
+      numTestRecords += 1
+
+  predictionAccuracy = 100.0 * numCorrect / numTestRecords
+
+  results = ("RESULTS: accuracy=%s | %s correctly predicted records out of %s "
+             "test records \n" % (predictionAccuracy,
+                                  numCorrect,
+                                  numTestRecords))
+  print results
+
+  return numCorrect, numTestRecords, predictionAccuracy
+
+
+
+def configureNetwork(inputFile,
+                     networkConfiguration):
+  """
+  Configure the network for various experiment values.
+  
+  @param inputFile (str) file containing the input data that will be fed to the 
+  network
+  @param networkConfiguration (dict) the configuration of this network.
+  """
+
+  # create the network encoders for sensor data.  
+  minval, maxval = _findCSVMinMax(inputFile)
+  _setScalarEncoderMinMax(networkConfiguration, minval, maxval)
+  encoders = networkConfiguration[SENSOR_REGION_NAME]["encoders"]
+
+  # Create and run network on this data.
+  #   Input data comes from a CSV file (scalar values, labels). The
+  #   RecordSensor region allows us to specify a file record stream as the
+  #   input source via the dataSource attribute.
+  dataSource = FileRecordStream(streamID=inputFile)
+
+  network = createNetwork(dataSource,
+                          encoders,
+                          networkConfiguration)
+
+  # Need to init the network before it can run.
+  network.initialize()
   return network
+
+
+
+def _findNumberOfPartitions(networkConfiguration, numRecords):
```

Comment:
correct. I made a change so that we are sure to test and train on the same length of data, regardless of the network configuration.